### PR TITLE
test: Add e2e keboola test testing background workers implementation

### PR DESCRIPTION
**Pull request by Matovidlo**
**Changes:**
- Add E2E test testing background workers implementation.
- Change `AssertKeys` to accept `wildcards`.

-----------

### CODE REVIEW

File: test/stream/bridge/keboola/keboola_test.go [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+		[]model.Slice{
+			{State: model.SliceWriting},
+			{State: model.SliceWriting},
+		},
+		[]model.Slice{
+			{State: model.SliceUploaded},
+			{State: model.SliceWriting},
+			{State: model.SliceUploaded},
+			{State: model.SliceWriting},
+		},
```
Comment (michaljurecko) [inline test/stream/bridge/keboola/keboola_test.go:82]: Some structure may be better, I don't know at first look, what the numbers mean.

**Review by michaljurecko — APPROVED** [review]

File: test/stream/bridge/keboola/keboola_test.go (hunk 1 above)
Comment (Matovidlo) [reply test/stream/bridge/keboola/keboola_test.go:82]: I know, could not think a better solution than this atm. Need to think it through more, but don't want to spend too much time for overengineering :sweat_smile:

Comment (Matovidlo) [reply test/stream/bridge/keboola/keboola_test.go:82]: https://github.com/keboola/keboola-as-code/pull/1983/commits/62321d71e871d3b0610c47833a0dc94ba57078ef

File: internal/pkg/service/common/dependencies/mocked.go [hunk 2]
```diff
@@ -309,8 +309,7 @@ func NewMocked(tb testing.TB, ctx context.Context, opts ...MockedOption) Mocked
 
 	// Clear logs
 	cfg.debugLogger.Truncate()
-
-	return d
+	return d, cfg.useRealAPIs
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/common/dependencies/mocked.go:312]: https://github.com/keboola/keboola-as-code/pull/1983/commits/358c87777b9db79f8134d8bf941ab5643c0a4362

**Pull request merged**